File: solutions/069.sqrtx/sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if(x == 0){
            return 0;
        }else if(x >= 1 && x < 4){
            return 1;
        }else if(x >=4 && x < 9){
            return 2;
        }
        for(int i = 3; i <= x/3; i++){
            if(x / i == i) {
                return i;
            }
            if(x / i > i && x /(i+1) < i+1){
                return i;
            }
        }
        return 0;
    }
}

File: solutions/069.sqrtx/sqrtx.js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    
    let left = 1, right = x;
    while(left < right){
        let mid = Math.floor(left + (right - left) / 2);
        if(mid**2 === x){
            return mid;
        }else if(mid**2 > x){
            right = mid;
        }else{
            left = mid + 1;
        }
    }
    if(left** 2 > x) return left - 1;
    return left;
};